src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -233,6 +233,8 @@ public function findOrNew($id, $columns = ['*'])
      */
     public function firstOrNew(array $attributes)
     {
+        $attributes = $this->model->newInstance()->fill($attributes)->attributesToArray();
```

Comment:
> not create a new model anyway because those attributes are guarded

Mass assignment exception doesn't occur when you call `save()`, it occur within `fill()` https://github.com/laravel/framework/blob/5.2/src/Illuminate/Database/Eloquent/Model.php#L439-L457

You might want to look at `forceFill()`.